dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/UploadStatus.kt

```diff
@@ -7,28 +7,39 @@
 package com.datadog.android.core.internal.data.upload
 
 import com.datadog.android.api.InternalLogger
+import java.util.Locale
 
 internal sealed class UploadStatus(
     val shouldRetry: Boolean = false,
     val code: Int = UNKNOWN_RESPONSE_CODE,
-    val throwable: Throwable? = null
+    val throwable: Throwable? = null,
+    val attempt: Int = 0
```

Comment:
While I agree this could be done in different ways the initial plan for me was to align to `iOS` implementation where this was added to the `UploadStatus` information, I also kept their naming -> `attempt`.